src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -1,119 +1,126 @@
-		c.weighty = 1;
-		content.add(btnClearCommand, c);
-
-		JButton btnClearObservers = new JButton("Delete observers");
-		btnClearObservers.addActionListener((ActionEvent e) -> this.deleteObservers());
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
-		content.add(btnClearObservers, c);
-	}
-
-	private void clearCommand() {
-		commandManager.clearCurrentCommand();
-		updateCurrentCommandField();
-	}
-
-	public void updateCurrentCommandField() {
-		currentCommandField.setText(commandManager.getCurrentCommandString());
-	}
-
-	public void deleteObservers() {
-		commandManager.getChangePublisher().clearObservers();
-		this.updateObserverListField();
-	}
-
-	private void updateObserverListField() {
-		observerListString = "";
-		List<Subscriber> commandChangeSubscribers = commandManager.getChangePublisher().getSubscribers();
-		for (Subscriber observer : commandChangeSubscribers) {
-			observerListString += observer.toString() + System.lineSeparator();
-		}
-		if (commandChangeSubscribers.isEmpty())
-			observerListString = "No observers loaded";
-
-		observerListField.setText(observerListString);
-	}
-
-	@Override
-	public void HideIfVisibleAndShowIfHidden() {
-		updateObserverListField();
-		if (this.isVisible()) {
-			this.setVisible(false);
-		} else {
-			this.setVisible(true);
-		}
-	}
+    private DriverCommandManager commandManager;
+
+    private JTextArea currentCommandField;
+
+    private String observerListString;
+    private JTextArea observerListField;
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 9204679248304669948L;
+
+    public CommandManagerWindow(DriverCommandManager commandManager) {
+        this.setTitle("Command Manager");
+        this.setSize(400, 400);
+        Container content = this.getContentPane();
+        content.setLayout(new GridBagLayout());
+
+        this.commandManager = commandManager;
+
+        GridBagConstraints c = new GridBagConstraints();
+
+        observerListField = new JTextArea("");
+        observerListField.setEditable(false);
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1;
+        c.gridx = 0;
+        c.weighty = 1;
+        content.add(observerListField, c);
+        updateObserverListField();
+
+        currentCommandField = new JTextArea("");
+        currentCommandField.setEditable(false);
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1;
+        c.gridx = 0;
+        c.weighty = 1;
+        content.add(currentCommandField, c);
+        updateCurrentCommandField();
+
+        // POLE DO WPISYWANIA KOMEND
+        JTextArea jTextArea = new JTextArea();
+        content.add(jTextArea, c);
+
+        JButton btnSendText = new JButton("Save figure to draw");
+        btnSendText.addActionListener((ActionEvent e) -> { //*************************** OBSLUGA BUTTONA **********************************
+
+            ISourceToDraw iSourceToDraw = new ListOfCommandsFromJTextArea();
```

Comment:
The interface should be used to hide implementation from CommandManagerWindow (view).
Instantiating using ListOfCommandsFromJTextArea ties view with implementation.
To fix it: an instance should be injected into view (similarly to commandManager)